### PR TITLE
chore: add detailed error messages for `--compat` flag verification

### PR DESCRIPTION
**Pull request by Sec-ant**
Addresses https://github.com/stackblitz-labs/pkg.pr.new/issues/104#issuecomment-2160112443

I'm not good at writing messages. Feel free to reword~

### CODE REVIEW

File: packages/cli/index.ts [hunk 1]
```diff
@@ -311,23 +308,32 @@ function hijackDeps(
...
-    if (!match) {
-      throw error;
-    }
-  } catch {
-    throw error;
+  const match = extractOwnerAndRepo(repository);
+  if (!match) {
+    throw new Error(
+      `pkg-pr-new cannot extract the owner and repo names from the ${packageName} repository link: ${repository}. --compact flag requires these names.
+Make sure to configure the 'repository' / 'repository.url' field in its package.json properly.
```
Comment (Aslemammad) [inline packages/cli/index.ts:335]: can we make this part that is present in the two error messages a static variable or something like that? so we reuse it! 

like `const instruction` in the top of the function

File: packages/utils/index.ts [hunk 2]
```diff
@@ -1,19 +1,18 @@
 import type { PackageManifest } from "query-registry";
 
 const githubUrlRegex =
-  /(?:git\+)?https?:\/\/github\.com\/([^\/]+\/[^\/]+)\.git/; // TODO: Don't trust this, it's chatgbd :)
+  /(?:git\+)?https?:\/\/github\.com\/([^\/]+)\/([^\/]+)\.git/; // TODO: Don't trust this, it's chatgbd :)
```
Comment (Aslemammad) [inline packages/utils/index.ts:4]: what does this change do?

File: packages/cli/index.ts (hunk 1 above)
Comment (Sec-ant) [reply packages/cli/index.ts]: Addressed in `fd461c3`.

File: packages/utils/index.ts (hunk 2 above)
Comment (Sec-ant) [reply packages/utils/index.ts:4]: Use two capture groups to capture the owner and repo so we don't need to `match[1].split("/")` later.

Comment (Aslemammad) [reply packages/utils/index.ts:4]: amazing!

**Review by Aslemammad — APPROVED** [review]

**Comment by Aslemammad** [comment]
Thank you so much for the effort!

**Pull request merged**